File: benchmarks/java/set/Java_Persistent/RangeSmall.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class RangeSmall {






    private BInteger counter;
    private BSet<BInteger> set;

    public RangeSmall() {
        counter = new BInteger(0);
        set = new BSet<>();
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000))).booleanValue()) {
            set = BSet.range(new BInteger(0),new BInteger(5));
            counter = counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        RangeSmall exec = new RangeSmall();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}

File: microbenchmarks/cpp/set/RangeSmall.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BSet.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef RangeSmall_H
#define RangeSmall_H

using namespace std;

class RangeSmall {



    private:



        BInteger counter;
        BSet<BInteger > set;

    public:

        RangeSmall() {
            counter = (BInteger(0));
            set = (BSet<BInteger >());
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))).booleanValue()) {
                set = (BSet<BInteger>::range((BInteger(0)),(BInteger(5))));
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    RangeSmall exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

